packages/vite/src/node/server/shortcuts.ts

```diff
@@ -2,30 +2,33 @@ import type { ViteDevServer } from '..'
 import { openBrowser, resolveBrowserUrl } from './openBrowser'
 import { restartServer } from './hmr'
 
-export function bindShortcuts(server: ViteDevServer, isRestart = false): void {
+export function bindShortcuts(server: ViteDevServer): void {
+  if (!server.httpServer) return
+
   server.config.logger.info(
     `  > Shortcuts: ` +
       SHORTCUTS.map((shortcut) => {
         return `"${shortcut.key}" ${shortcut.name}`
       }).join(', ')
   )
 
-  if (isRestart) return
-
-  const { stdin } = process
-  stdin.resume()
-  stdin.setEncoding('utf8')
-  stdin.setRawMode(true)
-  stdin.on('data', (data) => {
-    const input = data.toString().trim().toLowerCase()
```

Comment:
- removed `toString` call, since `setEncoding` ensures `data` is a string
- removed `trim` call, since whitespace is not a concern
- removed `toLowerCase` call, since we may want to use the uppercase shortcuts in the future